pandas/core/groupby.py

```diff
@@ -1380,6 +1380,32 @@ def cumsum(self, axis=0, *args, **kwargs):
 
     @Substitution(name='groupby')
     @Appender(_doc_template)
+    def rank(self, axis=0, method='average', numeric_only=True,
+             na_option='keep', ascending=True, pct=False):
+        """Compute numerical data ranks (1 through n) along axis.
+        """
+
+        def wrapper(values):
+            return values.rank(axis=axis, method=method, na_option=na_option,
+                               ascending=ascending, pct=pct)
```

Comment:
I don't understand what you are suggesting, exactly.

If I do
```
        if numeric_only:
            data = self._obj_with_exclusions._get_numeric_data()
            if data.size == 0:
                raise DataError('No numeric types to aggregate')
            data = data.groupby(self.grouper)
            return data.transform(wrapper)

        try:
            return self.transform(wrapper)
        except ValueError:
            if not numeric_only and method == 'first':
                raise ValueError('first not supported for non-numeric data')
                # such a ValueError is raised by pandas.algos.rank_2d_generic
                # for regular (non-grouped) dataframes
            raise
```
then the following test fails:
```
        df = DataFrame({'a': ['A1', 'A1', 'A1'],
                        'b': [datetime(2002, 2, 2), datetime(2001, 1, 1),
                              datetime(2001, 1, 1)],
                        'c': 1.})
        df = df.set_index('a')
        dg = df.groupby('c')

        result = dg.rank(method='first')
```
because `DataError('No numeric types to aggregate')` is raised